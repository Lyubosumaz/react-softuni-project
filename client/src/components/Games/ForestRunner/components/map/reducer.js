import { generalActions } from './actions';

const initialState = {
    tiles: [],
};

const mapReducer = (state = initialState, action) => {
    switch (action.type) {
        case generalActions.SET_TILES:
            return {
                ...action.payload,
            };
        default:
            return state;
    }
};

export default mapReducer;
