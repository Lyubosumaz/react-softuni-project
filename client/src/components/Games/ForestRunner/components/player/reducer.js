const initialState = {
    position: [0, 0],
    spriteLocation: '0px 0px',
    default: 'EAST',
    walkIndex: 0,
};

const playerReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'MOVE_PLAYER':
            return {
                ...state,
                ...action.payload,
            }
        case 'RESET_PLAYER_LOCATION':
            return {
                ...state,
                ...initialState
            }
        default:
            return state;
    };
};

export default playerReducer;