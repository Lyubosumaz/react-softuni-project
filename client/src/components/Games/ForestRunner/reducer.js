const initialState = {
    inGame: false,
    gold: 0,
    item: [],
    time: 0,
    handleGameTimer: false,
    level: 1,
    gameItems: [],
    inventorySellItem: false,
    inventoryEquipItem: false,
    characterRemoveItem: false,
};

const gameReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'SAVE_GAME_ITEMS_LOOT':
            return {
                ...state,
                gameItems: action.payload,
            };
        case 'GAME_TIMER_STATE':
            return {
                ...state,
                handleGameTimer: action.payload,
            };
        case 'SET_TIME':
            return {
                ...state,
                time: action.payload,
            };
        case 'OPEN_GOLD_CHEST':
            return {
                ...state,
                gold: (state.gold += action.payload),
            };
        case 'OPEN_ITEM_CHEST':
            return {
                ...state,
                item: state.item.concat(action.payload),
            };
        case 'START_THE_GAME':
            return {
                ...state,
                inGame: action.payload,
            };
        case 'END_THE_GAME':
            return {
                ...state,
                inGame: action.payload,
                level: 2,
            };
        case 'RESET_GAME_LEVEL':
            return {
                ...state,
                gold: 0,
                item: [],
                time: 0,
            };
        case 'CHARACTER_SELL_ITEM':
            return {
                ...state,
                inventorySellItem: action.payload,
            };
        case 'CHARACTER_EQUIP_ITEM':
            return {
                ...state,
                inventoryEquipItem: action.payload,
            };
        case 'CHARACTER_REMOVE_ITEM':
            return {
                ...state,
                characterRemoveItem: action.payload,
            };
        default:
            return state;
    }
};

export default gameReducer;
