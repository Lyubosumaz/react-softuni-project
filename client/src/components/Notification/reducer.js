import { generalActions } from './actions';
import numberGenerator from '../../utils/numberGenerator';

const initialState = {
    notifications: [],
};

const userReducer = (state = initialState, action) => {
    switch (action.type) {
        case generalActions.ADD_NEW_NOTIFICATION:
            return {
                ...state,
                notifications: [...state.notifications, { ...action.payload, id: '_id' + '__' + numberGenerator() + '__' + numberGenerator() }],
            };
        case generalActions.REMOVE_NOTIFICATION:
            return {
                ...state,
                notifications:
                    state.notifications[action.payload] ?
                    [...state.notifications.slice(0, action.payload), ...state.notifications.slice(action.payload + 1, state.notifications.length)]
                    : state.notifications,
            };
        case generalActions.REMOVE_ALL_NOTIFICATION:
            return {
                ...state,
                notifications: [],
            };
        default:
            return state;
    }
};

export default userReducer;
