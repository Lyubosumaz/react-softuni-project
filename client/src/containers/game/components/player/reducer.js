const initialState = {
    inGame: false,
    position: [0, 0],
    spriteLocation: '0px 0px',
    default: 'EAST',
    walkIndex: 0,
    final: false,
    gold: 0,
    items: []
};

const playerReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'MOVE_PLAYER':
            return {
                ...state,
                ...action.payload,
            }
        case 'OPEN_CHEST':
            return {
                ...state,
                gold: state.gold += action.payload,
            };
        case 'FINAL':
            return;
        case 'START':
            return {
                ...state,
                inGame: action.payload,
            }
        default:
            return state;
    };
};

export default playerReducer;