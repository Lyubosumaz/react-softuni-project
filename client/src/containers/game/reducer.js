const initialState = {
    inGame: false,
    final: false,
    gold: 0,
    item: [],
    time: 0,
    level: 1,
    gameItems: [],
    inventorySellItem: false,
    inventoryEquipItem: false,
    characterRemoveItem: false
};

const gameReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'SAVE_GAME_ITEMS':
            return {
                ...state,
                gameItems: action.payload,
            }
        case 'START':
            return {
                ...state,
                inGame: action.payload,
            }
        case 'OPEN_GOLD_CHEST':
            return {
                ...state,
                gold: state.gold += action.payload,
            };
        case 'OPEN_ITEM_CHEST':
            return {
                ...state,
                item: state.item.concat(action.payload),
            };
        case 'GET_TIME':
            return {
                ...state,
                time: action.payload,
            };
        case 'FINAL':
            return {
                ...state,
                inGame: action.payload,
                level: 2,
            };
        case 'RESET_GAME':
            return {
                ...state,
                final: false,
                gold: 0,
                item: [],
                time: 0,
                // level: 1
            };
        case 'CHARACTER_SELL_ITEM':
            return {
                ...state,
                inventorySellItem: action.payload
            };
        case 'CHARACTER_EQUIP_ITEM':
            return {
                ...state,
                inventoryEquipItem: action.payload
            };
        case 'CHARACTER_REMOVE_ITEM':
            return {
                ...state,
                characterRemoveItem: action.payload
            };
        default:
            return state;
    };
};

export default gameReducer;