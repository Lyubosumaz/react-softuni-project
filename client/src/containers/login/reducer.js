const initialState = {
    isLogin: false,
    userId: null,
}

const loginReducer = (state = initialState, action) => {
    switch (action.type) {
        case 'USER_LOGGED_IN':
            return {
                ...state,
                isLogin: true,
                userId: action.payload._id
            }
        default:
            return state;
    }
}

export default loginReducer;