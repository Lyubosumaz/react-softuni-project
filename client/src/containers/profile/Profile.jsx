import React, { useState, useEffect } from 'react';
import { connect } from 'react-redux';
import GameHistoryCard from './game-history-card/GameHistoryCard';
import handleRoute from '../../utils/handleRoutes';
import defaultProfilePic from '../../assets/img/default_profile.png';
import http from '../../services/http';
import './profile.css';

function Profile(props) {
    const [profile, setProfile] = useState(null);

    useEffect(() => {
        http.User.profile().then((gameProfile) => { setProfile(gameProfile); })
    }, []);

    console.log(profile)
    return (
        <div className="main-container">
            <h1>Profile</h1>

            {profile &&
                <div className="profile-card">
                    <div>
                        <h1>Name: {props.userName}</h1>
                        <img src={defaultProfilePic} alt="Profile" />
                    </div>

                    <div className="profile-stats">
                        <p><b>Your Game Profile Records:</b></p>
                        <p><b>Total Games Played: {profile.totalGames}</b></p>
                        <p><b>Total Time Played: {profile.totalTime}</b></p>
                        <p><b>Total Gold Accumulated: {profile.totalGames}</b></p>
                    </div>
                </div>}

            <div className="profile-game-history">
                <GameHistoryCard />
                <GameHistoryCard />
                <GameHistoryCard />
                <GameHistoryCard />
                <GameHistoryCard />
                <GameHistoryCard />
                <GameHistoryCard />
                <GameHistoryCard />
            </div>

            <div className="info-container">
                <p>Play one more game <button className="info-button" onClick={handleRoute('/game')}>Here</button>!</p>
            </div>
        </div >
    );
};

function mapStateToProps(state) {
    return {
        userName: state.user.userName,
    };
};

export default connect(mapStateToProps)(Profile);