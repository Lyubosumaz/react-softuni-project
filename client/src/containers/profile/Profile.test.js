import React from 'react';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { mount } from 'enzyme';
import reducer from '../game/reducer';
import Profile from './Profile';

describe('Profile Component', () => {
    let getWrapper;
    beforeEach(() => {
        const mockStore = createStore(reducer, { user: { userName: 'test' } });
        jest.mock('history', () => ({
            push: jest.fn(fn => fn()),
        }));
        getWrapper = () => mount(
            <Provider store={mockStore}>
                <Profile />
            </Provider>
        );
    });

    it('Renter Profile', () => {
        const wrapper = getWrapper();
        expect(wrapper).toBeTruthy();
    });

    it('Should render without errors main-container', () => {
        const wrapper = getWrapper().find('.main-container');
        expect(wrapper.length).toBe(1);
    });

    it('Checks for html changes', () => {
        const wrapper = getWrapper();
        expect(wrapper).toMatchSnapshot();
    });
});
