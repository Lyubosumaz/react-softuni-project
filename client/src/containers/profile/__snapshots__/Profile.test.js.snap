// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Component Checks for html changes 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Profile)>
    <Profile
      dispatch={[Function]}
      userName="test"
    >
      <div
        className="main-container"
      >
        <h1>
          Profile
        </h1>
        <div
          className="profile-game-history"
        />
        <div
          className="info-container"
        >
          <p>
            Play one more game 
            <button
              className="info-button"
              onClick={[Function]}
            >
              Here
            </button>
            !
          </p>
        </div>
      </div>
    </Profile>
  </Connect(Profile)>
</Provider>
`;
